src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -343,38 +343,33 @@ class IpexCommunicationService extends AgentService {
     });
   }
 
+  private async getSchema(schemaSaid: string) {
+    try {
+      const schema = await this.props.signifyClient.schemas().get(schemaSaid);
+      return schema;
+    } catch (error) {
+      const errorStack = (error as Error).stack as string;
+      const status = errorStack.split("-")[1];
+      if (/404/gi.test(status) && /SignifyClient/gi.test(errorStack)) {
+        const oobi = await Agent.agent.connections.resolveOobi(
+          `${ConfigurationService.env.keri.credentials.testServer.urlInt}/oobi/${schemaSaid}`
+        );
+        if (oobi.done) {
+          await this.getSchema(schemaSaid);
```

Comment:
Just in case there is a bug with signify (or an issue with OOBIs, since an OOBI can be used for more than just schemas) - can we add a base case to this recursion?

So `schemaSaid: string, retry = true` and call `await this.getSchema(schemaSaid, false)` on this line. So we just try once